### ISSUE TITLE
⌨ Dismissible chips improperly nested

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#6762]
## ⌨ Accessibility Issue

Our dismissible chips appear in a `ul` but are nested improperly as `list > button > listitem`.

### WCAG 2.1 Criteria

[Certain ARIA roles must contain particular children.](https://dequeuniversity.com/rules/axe/4.7/aria-required-children?application=axeAPI)

## 🦋 Expected Behaviour

The chips should be nested so that the `listitem` is a direct child of the parent `list`.

## 📋 Steps to Reproduce

1. Open the "Chip multi" story
2. Confirm the items are `listitem`s wrapped by `button`s and not the inverse

## 📸 Screenshot

![Screenshot 2023-05-31 092408](https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/c693f92c-23cf-4685-9bd4-161379aaa1f9)

## 🙋‍♀️ Proposed Solution

> **Warning**
> This could be a bit tricky!...or not 🤷‍♀️ 

- [ ] Update dismissible chips to to be nested as `listitem > button`

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Multiple, dismissible chips have the following HTML structure `list > listitem > button`
- [ ] No differences in the look and feel
